src/Avalonia.Controls/Control.cs

```diff
@@ -309,5 +310,65 @@ protected override void OnKeyUp(KeyEventArgs e)
                 }
             }
         }
+
+        protected override void OnPropertyChanged<T>(AvaloniaPropertyChangedEventArgs<T> change)
+        {
+            base.OnPropertyChanged(change);
+            
+            if (change.Property == FlowDirectionProperty)
+            {
+                // A change in value inherited should be prevented from calling this method
+                // because it will be handled from here
+                if (GetBaseValue(FlowDirectionProperty, change.Priority).HasValue)
+                {
+                    InvalidateFlowDirection();
+                    
+                    foreach (var visual in this.GetVisualDescendants())
+                    {
+                        if (visual is Control child)
+                        {
+                            child.InvalidateFlowDirection();
+                        }
+                    }
+
+                    InvalidateVisual();
+                }
+            }
+        }
+
+        /// <summary>
+        /// Computes the <see cref="IVisual.IsMirrorTransform"/> value according to the 
+        /// <see cref="FlowDirection"/>
+        /// </summary>
+        public virtual void InvalidateFlowDirection()
+        {
+            bool parentShouldPresentedMirrored = false;
+            bool thisShouldPresentedMirrored = ShouldPresentedMirrored();
+
+            var parent = this.FindAncestorOfType<Control>();
+            if (parent != null)
+            {
+                parentShouldPresentedMirrored = parent.ShouldPresentedMirrored();
+            }
+            else if (this.Parent is Control logicalParent)
+            {
+                parentShouldPresentedMirrored = logicalParent.ShouldPresentedMirrored();
+            }
+
+            bool shouldApplyMirrorTransform = thisShouldPresentedMirrored != parentShouldPresentedMirrored;
+
+            IsMirrorTransform = shouldApplyMirrorTransform;
+        }
+
+        /// <summary>
+        /// Determines whether the element should be presented mirrored, this
+        /// method related to FlowDirection system and returns true if FlowDirection
+        /// is RightToLeft. For controls that want to avoid this behavior, it is 
+        /// possible to override this method and return false. 
+        /// </summary>
+        protected virtual bool ShouldPresentedMirrored()
```

Comment:
`ShouldPresentMirrored` is correct English. This applies a few different places.

Reply:
So, what do you mean?

Reply:
It should be fixed since this is API. It needs to be changed in all places in the code.

Reply:
The larger debate about naming itself I leave to the core team.

Reply:
maybe `CanBeMirrored`?

Reply:
its not my place to say if the name should change that much. what I am saying is "ShouldPresentedMirrored" -> "ShouldPresentMirrored" for proper English.

Reply:
oops, now I understand what you mean, I did not notice `ed `